core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -794,6 +797,45 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
   }
 
   /**
+   * Reads in a directory of Avro files from HDFS, a local file system (available on all nodes), or
+   * any Hadoop-supported file system URI. The records are read in as Generic Avro records. This
+   * also allows a user to register a schema with Kryo, if they so choose to.
+   *
+   * You can do the following if you know the schema ahead of time:
+   * {{{
+   *   val schema = new Schema.Parser().parse(schemaString)
+   *   sc.avroFile("/input-path", schema)
+   * }}}
+   *
+   * or just:
+   * {{{
+   *   sc.avroFile("/input-path")
+   * }}}
+   */
+  def avroFile(path: String, schemas: Schema*): RDD[GenericRecord] = {
```

Comment:
Not quite familiar with Avro, but why do we need to pass in more than one `Schema` instances here? Is this because all nested `Schema` classes must also be registered? If it is the case, would be nice to document it.